src/py21cmfast/wrapper.py

```diff
@@ -1256,6 +1345,9 @@ def ionize_box(
         If given, this field will be used, otherwise it will be generated. To be generated, either `init_boxes` and
         `redshift` must be given, or `user_params`, `cosmo_params` and `redshift`.
 
+    previous_perturbed_field : :class:`~PerturbField`, optional
+        An perturbed field at higher redshift. This is only used if mini_halo is included.
```

Comment:
You have added the previous_perturbed_field argument and added it to ionised box.

However, in the run_lightcone and run_coeval classes etc. (and possibly other functions) you haven't added the previous_perturbed_field argument to the calculation.